internal/blockchain/indexers/indexsubscriber.go

```diff
@@ -412,7 +397,20 @@ func (s *IndexSubscriber) Run(ctx context.Context) {
 	s.wg.Add(2)
 	go s.handleIndexUpdates(ctx)
 	go s.handleSyncSubscribers(ctx)
-	s.wg.Wait()
 
-	log.Infof("Index subscriber shutting down")
+	// Stop all the subscriptions and shutdown the subscriber when the context
+	// is cancelled.
+	<-ctx.Done()
+	close(s.quit)
+	s.mtx.Lock()
+	for _, sub := range s.subscriptions {
+		err := sub.stop()
+		if err != nil {
+			log.Error("unable to stop index subscription: %v", err)
```

Comment:
Not introduced by this PR, but this should be `Errorf` and the message capitalized.

Reply:
Indeed.  Updated.